integration_test/workflow_tests/test_packages/ingest/test_no_granules.py

```diff
@@ -0,0 +1,95 @@
+
+import boto3
+
+import helpers
+from requests import Session
+
+
+class TestNoGranules(TestCase):
+    def test_no_granules_passes(self):
+        """
+        If no granules are provided, should not store anything in DB.
+        """
+        # Set up Orca API resources
+        # ---
+        my_session = Session()
+        my_session.mount(
+            helpers.api_url,
+            helpers.DNSResolverHTTPSAdapter(
+                helpers.aws_api_name, helpers.API_LOCAL_HOST
+            ),
+        )
+
+        provider_id = uuid.uuid4().__str__()
+        provider_name = uuid.uuid4().__str__()
+
+        collection_name = uuid.uuid4().__str__()
+        collection_version = uuid.uuid4().__str__()
+
+        execution_id = uuid.uuid4().__str__()
+
+        # note that this does not cover all available properties
+        copy_to_glacier_input = {
+            "payload": {"granules": []},
+            "meta": {
+                "collection": {"name": collection_name, "version": collection_version},
+                "provider": {
+                    "id": provider_id,
+                    "name": provider_name,
+                },
+            },
+            "cumulus_meta": {"execution_name": execution_id},
+        }
+
+        expected_output = {
+            "payload": {"granules": [], "copied_to_glacier": []},
+            "meta": {
+                "collection": {"name": collection_name, "version": collection_version},
+                "provider": {"id": provider_id, "name": provider_name},
+            },
+            "cumulus_meta": {"execution_name": execution_id},
+            "task_config": {
+                "excludeFileTypes": "{$.meta.collection.meta.excludeFileTypes}",
+                "s3MultipartChunksizeMb": "{$.meta.collection.meta.s3MultipartChunksizeMb}",
+                "providerId": "{$.meta.provider.id}",
+                "providerName": "{$.meta.provider.name}",
+                "executionId": "{$.cumulus_meta.execution_name}",
+                "collectionShortname": "{$.meta.collection.name}",
+                "collectionVersion": "{$.meta.collection.version}",
+                "orcaDefaultBucketOverride": "{$.meta.collection.meta.orcaDefaultBucketOverride}",
+                "orcaDefaultStorageClassOverride":
+                    "{$.meta.collection.meta.orcaDefaultStorageClassOverride}",
+            },
+            "exception": "None",
+        }
+
+        execution_info = boto3.client("stepfunctions").start_execution(
+            stateMachineArn=helpers.orca_copy_to_glacier_step_function_arn,
+            input=json.dumps(copy_to_glacier_input, indent=4),
+        )
+
+        step_function_results = helpers.get_state_machine_execution_results(
+            execution_info["executionArn"]
+        )
+
+        self.assertEqual("SUCCEEDED", step_function_results["status"])
+        self.assertEqual(expected_output, json.loads(step_function_results["output"]))
+
+        catalog_output = helpers.post_to_api(
```

Comment:
A couple comments may be good just to explain that the first set of asserts verify the step function integration works and the second set checks that it made it to the database.

One downside to this check is that we do not really know if it is valid or not since sending a bogus providerId would result in the same output. We may want to consider this in the future.

Reply:
Added